docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -0,0 +1,604 @@
+      - ["As-is" startup flow](#-as-is--startup-flow)
+      - ["To-be" startup flow](#-to-be--startup-flow)
+      - [install](#install)
+      - [Vault](#vault)
+      - [Consul](#consul-1)
+      - [PostgreSQL](#postgresql)
+      - [Redis](#redis)
+      - [vault-worker](#vault-worker)
+      - [ready_to_run](#ready-to-run)
+      - [Kong](#kong)
+      - [All other (stateless) services](#all-other--stateless--services)
+  * [Consequences](#consequences)
+    + [Benefits](#benefits)
+    + [Drawbacks](#drawbacks)
+  * [Alterantives](#alterantives)
+    + [Event-driven vs commanded staging](#event-driven-vs-commanded-staging)
+    + [System management agent (SMA) as the coordinator](#system-management-agent--sma--as-the-coordinator)
+    + [Separate compose scripts for initialization and runtime](#separate-compose-scripts-for-initialization-and-runtime)
+    + [Manual secret provisioning](#manual-secret-provisioning)
+  * [Phased implementation approach](#phased-implementation-approach)
+  * [References](#references)
+
+## Status
+
+** proposed **
+
+## Context
+
+This ADR seeks to accomplish two objectives in support of a greater
+objective of improving microservice communication security:
+
+- Define the recommended bootstrapping flow for security-enabled EdgeX
+- Define a Docker-compatible bootstrapping mechanism.
+  (This mechanism is expected to be compatible with a
+  single-pod Kubernetes deployment using hostPath persistent volumes.)
+
+A security-disabled installation of EdgeX is not required to follow this guidance.
+
+Ideally, adoption of this ADR would permit the Docker-based install
+flow and the snap-based install flow to share a common set of
+initializtion scripts.
+
+This ADR presumes prior approval of
+[ADR 0015 - Encryption between microservices](0015-Use-of-encryption-for-in-cluster-communication.md),
+which clarifies that TLS assets are not being bootstrapped as part of this procedure.
+
+### History
+
+In every EdgeX release from Barcelona to Hanoi, a foundational component, Consul,
+has been the first EdgeX microservice to start.
+Consul provides both a service registry and an HTTP-accessible configuration service.
+EdgeX services rely on Consul for:
+
+- Service location data (what host, port, and protocol is used to access another service)
+- Service health checks (used to stage gate security in Geneva and Hanoi)
+- General configuration data (i.e. not service location)
+- Storage engine for the EdgeX secret store
+- Fallback secret storage in the security-disabled mode of EdgeX
+
+Regrettably, Consul is not configured in a secure mode
+but instead allows anonymous read/write access to the entire data store
+by any adversary that can obtain access its REST API endpoint.
+Securing Consul is a major sub-objective of this ADR.
+
+We need a bootstrapping mechanism that ensures that all services,
+including Conul, are brought up in a secure configuration,
```

Comment:
IMO, a simpler approach to securing Consul is via network isolation in the secure compose file. Same for Kong Admin port. Don't expose them outside the docker network.

Reply:
IMO network segmentation is a defense of last resort.  We should still require a credential to talk to Consul